src/js/server/watcher.js

```diff
@@ -0,0 +1,102 @@
+/*
+
+    A "watcher" that looks for changes in one or more directories and fires an `onFsChange` event if:
+
+    1. Existing files change.
+    2. Files are added.
+    3. Files are removed.
+    4. Files are renamed.
+
+    Based on chokidar: https://github.com/paulmillr/chokidar
+
+ */
+// TODO: Discuss (with Antranig) moving this to its own micro-module if it's more broadly useful.
```

Comment:
Sounds like a good idea, assuming that build tools eventually arise that make working with micromodules substantially less arduous : P